components/mapping/default/index.mjs

```diff
@@ -98,37 +97,31 @@ export const createMapping = ({ url: base, content }) => {
   return {
     type: "normal",
     base,
-    sources: relatives
-      .map((relative) => toAbsoluteUrl(relative, root_base))
-      .map(
-        contents === null
-          ? (url) => ({ url, content: null })
-          : (url, index) => ({
-              url,
-              content: index < contents.length ? contents[index] : null,
-            }),
-      )
-      .map(addHashToFile),
+    sources: relatives.map((relative, index) =>
+      createSource(
+        toAbsoluteUrl(relative, root_base),
+        contents === null || contents === undefined || index >= contents.length
+          ? null
+          : contents[index],
+      ),
+    ),
     lines: parseGroupArray(mappings),
   };
 };
 
 export const mapSource = (mapping, line, column) => {
   if (mapping.type === "mirror") {
-    const { hash, url } = mapping.source;
-    return { hash, url, line, column };
+    return makeSourceLocation(mapping.source, line, column);
   } else if (mapping.type === "normal") {
     if (line <= mapping.lines.length) {
       for (const fields of mapping.lines[line - 1]) {
         if (fields[0] === column && fields.length >= 4) {
           if (fields[1] < mapping.sources.length) {
-            const { hash, url } = mapping.sources[fields[1]];
-            return {
-              hash,
-              url,
-              line: fields[2] + 1,
-              column: fields[3],
-            };
+            return makeSourceLocation(
+              mapping.sources[fields[1]],
+              fields[2] + 1,
+              fields[3],
+            );
```

Comment:
This could be clearer if you deconstructed the `fields` array.

Reply:
Rebased in b62efb0935dd303164a844d6b89b590e365d6a10